src/main/java/com/savitoh/casacodigoapi/validator/AutorCustomValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.savitoh.casacodigoapi.validator;
+
+import com.savitoh.casacodigoapi.exception.data.CreateAutorValidateException;
+import com.savitoh.casacodigoapi.model.Autor;
+import com.savitoh.casacodigoapi.repository.AutorRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+
+@Component
+public class AutorCustomValidator  {
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Opa, o Spring já possui o Spring Validation para possibilitar integrar validações customizadas no fluxo natural. Dessa forma você não precisa criar exceptions específicas  e mantém o código inteiro mais simples. Você pode dar uma olhada a partir daqui => https://github.com/asouza/youtube-apicdc/blob/master/src/main/java/br/com/deveficiente/youtubeapicdc/detalhelivro/CrudCategoriasController.java

Reply:
E o custom bean validation com as anotações customizadas de validação, acha overengineering?

Reply:
Opa, aqui é mais questão de conforto. Uso as validações do Spring pq a partir dali já tenho acesso a tudo que preciso do framework. Se eu quiser receber argumento no construtor etc...